### DISCUSSION TITLE
How can I refer a local file schema instead of introspection and SDL endpoint usage when using the gradle plugin?

### DISCUSSION THREAD

**Discussion by xetra11** [ExpediaGroup/graphql-kotlin#905] [Q&A] [answered]
The documentation says:

> GraphQL schema can be provided as
> 
> - result of introspection query (default)
> - downloaded from an SDL endpoint
> - local file

I wonder what I need to do in the gradle plugin to allow the `local file` way?
If I omit the `sdlEndpoint` in the 
```
graphql {
   client {
        ...
    }
}
```
configuration I receive the following error
```
Invalid GraphQL client extension configuration - missing required endpoint/sdlEndpoint property
```
But without omitting `sdlEndpoint` how can I tell the plugin to use my local file under `src/main/resources/graphql/schema.graphql` ?

**Comment by dariuszkuc** [comment, marked as answer]
`graphql` extension will apply introspection/sdl task to obtain the schema and use the resulting schema as an input to generate the client. It is not possible to use the extension to run just client generation. See [documentation](https://expediagroup.github.io/graphql-kotlin/docs/plugins/gradle-plugin#complete-configuration-example).

Instead you will need to configure the corresponding task directly, e.g.

```kotlin
val graphqlGenerateClient by tasks.getting(GraphQLGenerateClientTask::class) {
  packageName.set("com.example.generated")
  schemaFile.set(file("path/to/your/schema/file")
}
```

You can then invoke the corresponding task directly (it will also automatically add itself as a dependency of `compileKotlin` task)

```
./gradlew graphqlGenerateClient
```
